### PR TITLE
4 created placeholder functions in sdk interface

### PR DESCRIPTION
**Pull request by LuisOsta**

### CODE REVIEW

File: credentials.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+func (c *knoxClient) SharePresentation(params SharePresentationParams) error {
+	creds := params.Credentials
+	challenge, err := c.ca.CreatePresentationChallenge()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Challenge: %s, %s\n", challenge.Nonce, challenge.Url)
+	converted_creds := make([]map[string]interface{}, len(creds))
```
Comment (sKudryashov) [inline credentials.go:56]: variable name `convertedCreds` probably

File: credentials.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	}
+
+	proc := ld.NewJsonLdProcessor()
+	options := ld.NewJsonLdOptions("")
+	options.Format = NormalizationFormat
+	options.Algorithm = NormalizationAlgo
+	normalized, err := proc.Normalize(vp, options)
+
+	if err != nil {
+		fmt.Printf("Error normalizing: %s\n", err.Error())
```
Comment (sKudryashov) [inline credentials.go:80]: We have to either output an error or return it, not both

File: identity.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package knox
+
+import "errors"
+
+type RegisterIdentityParams struct {
+}
+
+type GenerateIdentityParams struct{}
```
Comment (sKudryashov) [inline identity.go:8]: please format both the same way

File: knox_test.go [hunk 4]
```diff
@@ -28,9 +30,9 @@ type requestCredentialTest struct {
 func TestRequestCredential(t *testing.T) {
 	cred_type := "BankCard"
 	mock_controller := gomock.NewController(t)
-	mock_wallet := knox_mock.NewMockWallet(mock_controller)
+	mock_wallet := s_mock.NewMockDynamicSigner(mock_controller)
```
Comment (sKudryashov) [inline knox_test.go:33]: variable and package name mock_wallet ans s_mock are underscored, we agreed on camelCase

**Review by sKudryashov — CHANGES_REQUESTED** [review]

File: knox_test.go (hunk 4 above)
Comment (LuisOsta) [reply knox_test.go:33]: The naming conventions for packages are either camel case or underscore - https://github.com/knox-networks/gomono

File: identity.go (hunk 3 above)
Comment (LuisOsta) [reply identity.go:8]: I'll do that

File: credentials.go (hunk 2 above)
Comment (LuisOsta) [reply credentials.go:80]: I'll remove it

File: credentials.go (hunk 1 above)
Comment (LuisOsta) [reply credentials.go]: Done

File: knox_test.go (hunk 4 above)
Comment (LuisOsta) [reply knox_test.go:33]: I think its better to have package names be lowercase (and underscore if necessary). I don't think camel case works for packages

**Review by sKudryashov — APPROVED** [review]

**Pull request merged**